### PR TITLE
[CRT-103] Added CI Testing

### PR DESCRIPTION
**Pull request by JoshGalantai-Vidyard**
https://vidyard.atlassian.net/browse/CRT-103

**Description**

Added some CI steps to run rspecs and do linting with rubocop.
(Using the same lint rules as implemented in alchemist: https://github.com/Vidyard/Alchemist/blob/master/.rubocop.yml)

Initially adding the lint threw many hundreds of violations, so for now a todo has been generated to address these via
`rubocop --auto-gen-config --exclude-limit 1000`

### CODE REVIEW

File: spec/ffmpeg/movie_spec.rb [hunk 1]
```diff
@@ -277,7 +277,7 @@ module FFMPEG
         end
 
         it "should parse the bitrate" do
-          expect(@movie.bitrate).to eq(481836)
+          expect(@movie.bitrate).to be_between(481736, 481936)
```
Comment (JoshGalantai-Vidyard) [inline spec/ffmpeg/movie_spec.rb:280]: This test appears to be flakey, bitrate isn't always exactly the same so we just require an approximation.

File: .github/workflows/ci.yaml [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+          bundle exec rspec
+  lint:
+    runs-on: ubuntu-latest
+    container: public.ecr.aws/docker/library/ruby:3.2.2
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v3
+      - name: Install Deps
+        run: |
+          apt update && apt install curl -y
```
Comment (Kyrluckechuck) [inline .github/workflows/ci.yaml]: nit: Shouldn't need this apt step iirc 🤔

File: .github/workflows/ci.yaml [hunk 3]
```diff
@@ -0,0 +1,33 @@
...
+jobs:
+  rspec:
+    runs-on: ubuntu-latest
+    container: public.ecr.aws/docker/library/ruby:3.2.2
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v3
+      - name: Install Deps
+        run: |
+          apt update && apt install curl -y
```
Comment (Kyrluckechuck) [inline .github/workflows/ci.yaml]: Could either move this `apt update && apt install ffmpeg -y` to the following step, or just replace the `curl -y` with `ffmpeg -y`!

**Review by Kyrluckechuck — DISMISSED** [review]
Just a couple of minor comments but looks good!

File: .github/workflows/ci.yaml (hunk 2 above)
Comment (JoshGalantai-Vidyard) [reply .github/workflows/ci.yaml]: fixed

**Review by Kyrluckechuck — APPROVED** [review]

File: .github/workflows/ci.yaml (hunk 3 above)
Comment (JoshGalantai-Vidyard) [reply .github/workflows/ci.yaml]: fixed

**Pull request merged**